_episodes/20-crop-raster-data.md

```diff
@@ -0,0 +1,218 @@
+---
+
+It is quite common that the raster data you have in hand is too large to process, or not all the pixels are relevant to your area of interest (AoI). In both situations, you should consider cropping your raster data before performing data analysis. 
+
+In this episode, we will introduce how to crop raster data into the desired area. We will use one Sentinel-2 image over Amsterdam as the example raster data, and introduce how to crop your data to different types of AoIs.
+
+> ## Introduce the Data
+>
+> Raster data: A Sentinel-2 raster image of Amsterdams: `S2_amsterdam.tif`.
+> 
+> Vector data: we will use three different types of vector data as AoIs:
+> - Crop field polygons in north Amsterdam. ([source](https://www.pdok.nl/introductie/-/article/basisregistratie-gewaspercelen-brp-)): `data/crop_fields`.
+> - Dike polylines in north Amsterdam. ([source](https://www.pdok.nl/downloads/-/article/basisregistratie-ondergrond-bro-#fa90454e447b478fb2db187bb6fc8a10)): `data/dikes`.
+> -Groud water monitoring wells in north Amsterdam. ([source](https://www.pdok.nl/downloads/-/article/basisregistratie-ondergrond-bro-#3f9edb6734c11af4886cdb37b69711bc)): `data/groundwater_monitoring_well`.
+{: .callout}
+
+## Crop raster data with a bounding box
+
+First, we can have a glimpse of the raster data by loading it with `rioxarray`:
+
+~~~
+import rioxarray
+
+# Load image and visualize
+raster = rioxarray.open_rasterio('data/S2_amsterdam.tif')
+raster.plot.imshow(figsize=(8,8))
+~~~
+{: .language-python}
+
+<img src="../fig/20-crop-raster-original-raster-00.png" title="Overview of the raster"  width="512" style="display: block; margin: auto;" />
+
+The raster data is quite big. It even takes tens of seconds to visualize. But do we  need the entire raster? Suppose we are interested in the crop fields, we can simply compare its coverage with the raster coverage:
+
+~~~
+import geopandas as gpd
+from shapely.geometry import box
+from matplotlib import pyplot as plt
+
+# Load the polygons of the crop fields
+cf_boundary_crop = gpd.read_file("data/crop_fields/cf_boundary_crop.shp")
+cf_boundary_crop = cf_boundary_crop.to_crs(raster.rio.crs) # convert to the same CRS
+
+# Plot the bounding box over the raster
+bounds = box(*cf_boundary_crop.total_bounds)
+bb_cropfields = gpd.GeoDataFrame(index=[0], crs=raster.rio.crs, geometry=[bounds])
+fig, ax = plt.subplots()
+fig.set_size_inches((8,8))
+raster.plot.imshow(ax=ax)
+bb_cropfields.plot(ax=ax, alpha=0.6)
+~~~
+{: .language-python}
+
+<img src="../fig/20-crop-raster-bounding-box-01.png" title="Bounding boxes of AoI over the raster"  width="512" style="display: block; margin: auto;" />
+
+Seeing from the bounding boxes, the crop fields (red) only takes a small part of the raster (blue). Therefore before actual processing, we can first crop the raster to the actual area of interest. The `clip_box` function allows one to crop a raster by the min/max of the x and y coordinates. 
```

Comment:
I don't see red :thinking: please note that some people have difficulties seeing colors on screen. Can we use `non-filled` boxes for both raster bounding box and field polygons with different borders such as solid and dashed lines?

Reply:
Indeed. Since we are going to update the source data,  I will solve this one later. This comment has been documented as an [issue](https://github.com/esciencecenter-digital-skills/geospatial-python/issues/1) in the forked repo